sdk/kiosk/src/tx/transfer-request.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { TransactionArgument, TransactionBlock } from '@mysten/sui.js';
+import { ObjectArgument, objArg } from '../utils';
+
+/**
+ * A Hot Potato struct that is returned in the `kiosk::purchase` function.
+ * Can only be consumed by the `transfer_policy::confirm_request`.
+ */
+export type TransferRequest = {
+  kind: 'NestedResult';
+  index: number;
+  resultIndex: number;
+  itemType: string;
+};
+
+/**
+ * Call the `transfer_policy::confirm_request` function to unblock the
+ * transaction.
+ */
+export function confirmRequest(
+  tx: TransactionBlock,
+  policy: ObjectArgument,
+  request: TransferRequest,
+): TransactionBlock {
+  tx.moveCall({
```

Comment:
Maybe return this instead of the tx so that the result values could be used, even if there's no meaningful result just for consistency.

Reply:
Or return undefined or null maybe